app/uk/gov/hmrc/helptosavefrontend/controllers/AccountHolderUpdateEmailAddressController.scala

```diff
@@ -56,20 +56,25 @@ class AccountHolderUpdateEmailAddressController @Inject() (val helpToSaveService
     )
   }(redirectOnLoginURL = routes.AccountHolderUpdateEmailAddressController.getUpdateYourEmailAddress().url)
 
-  def onSubmit(): Action[AnyContent] = authorisedForHtsWithNINO { implicit request ⇒ implicit htsContext ⇒
-    checkIfAlreadyEnrolled(_ ⇒
-      UpdateEmailForm.verifyEmailForm.bindFromRequest().fold(
-        formWithErrors ⇒ {
-          BadRequest(views.html.email.update_email_address("errors", formWithErrors))
-        },
-        (details: UpdateEmail) ⇒
-          sendEmailVerificationRequest(
-            details.email,
-            Ok(views.html.email.check_your_email(details.email)),
-            params ⇒ routes.AccountHolderUpdateEmailAddressController.emailVerified(params.encode()).url,
-            isNewApplicant = false)
-
-      ))
+  def onSubmit(): Action[AnyContent] = authorisedForHtsWithNINOAndName { implicit request ⇒ implicit htsContext ⇒
+    htsContext.name.fold[Future[Result]](
```

Comment:
if the `name` doesn't exist, isn't it right to show the user the missing user details page ? 
ideally `name` cannot be missing as its verified as part of IV checks , so not an issue

Reply:
This is an account holder who for some reason does not have a name - this is unlikely to happen since they could not have gone through the registration process without having a name. But when there is an error in updating an email for account holders there's this page which says "Update was unsuccessful, click here to go back to managing your account". That seemed like the most appropriate course to take here, but I'll make sure with Nibras